File: test/rs/case03_if_else.rs
#[path = "../../src/rs_module/py_runtime.rs"]
mod py_runtime;
use py_runtime::{perf_counter, py_print};

// このファイルは自動生成です（native Rust mode）。

fn abs_like(n: i64) -> i64 {
    if ((n) < (0)) {
        return (-n);
    } else {
        return n;
    }
}

fn main() {
    py_print(abs_like((-12)));
}

File: test/cpp/case03_if_else.cpp
// このファイルは `test/py/case03_if_else.py` のテスト/実装コードです。
// 役割が分かりやすいように、読み手向けの説明コメントを付与しています。
// 変更時は、既存仕様との整合性とテスト結果を必ず確認してください。


#include "cpp_module/py_runtime.h"

// このファイルは `test/py/case03_if_else.py` のテスト/実装コードです。
// 役割が分かりやすいように、読み手向けの説明コメントを付与しています。
// 変更時は、既存仕様との整合性とテスト結果を必ず確認してください。

int64 abs_like(int64 n) {
    if (n < 0)
        return -n;
    else
        return n;
}

int main() {
    py_print(abs_like(-12));
    return 0;
}
